### PR TITLE
Core/LOS: restruct LOS functions and add LineOfSight check 

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
- restruct LOS functions and add LineOfSight check

### CODE REVIEW

**Comment by Viste** [comment]
waits for https://github.com/azerothcore/azerothcore-wotlk/pull/1235

**Comment by BarbzYHOOL** [comment]
Awesome, can't say much more :P

**Comment by Viste** [comment]
redy for test

**Comment by Viste** [comment]
ready for tests

**Comment by sucofog** [comment]
Does Sindragosa Tomb get affected by checks?

https://github.com/azerothcore/azerothcore-wotlk/blob/832bccd9fa8ae5f3dd20eaeed244e74cd700f2de/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp#L1292

**Comment by Viste** [comment]
> Does Sindragosa Tomb get affected by checks?
> 
> [azerothcore-wotlk/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp](https://github.com/azerothcore/azerothcore-wotlk/blob/832bccd9fa8ae5f3dd20eaeed244e74cd700f2de/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp#L1292)
> 
> Line 1292 in [832bccd](/azerothcore/azerothcore-wotlk/commit/832bccd9fa8ae5f3dd20eaeed244e74cd700f2de)
>  return !dTree.isInLineOfSight(unit->GetPositionX(), unit->GetPositionY(), unit->GetPositionZ()+2.f, ox, oy, oz+2.f, unit->GetPhaseMask());

no here must be cheked only dynamic objects and if i add cheks here all will be break

**Comment by sucofog** [comment]
> > Does Sindragosa Tomb get affected by checks?
> > [azerothcore-wotlk/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp](https://github.com/azerothcore/azerothcore-wotlk/blob/832bccd9fa8ae5f3dd20eaeed244e74cd700f2de/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp#L1292)
> > Line 1292 in [832bccd](/azerothcore/azerothcore-wotlk/commit/832bccd9fa8ae5f3dd20eaeed244e74cd700f2de)
> > return !dTree.isInLineOfSight(unit->GetPositionX(), unit->GetPositionY(), unit->GetPositionZ()+2.f, ox, oy, oz+2.f, unit->GetPhaseMask());
> 
> no here must be cheked only dynamic objects and if i add cheks here all will be break
> you can look trinity don't add cheks for syndra to

Yes, i see now thanks for info

File: src/server/game/Maps/Map.cpp [hunk 1]
```diff
@@ -2164,10 +2164,16 @@ float Map::GetWaterLevel(float x, float y) const
         return 0;
 }
 
-bool Map::isInLineOfSight(float x1, float y1, float z1, float x2, float y2, float z2, uint32 phasemask) const
+bool Map::isInLineOfSight(float x1, float y1, float z1, float x2, float y2, float z2, uint32 phasemask, LineOfSightChecks checks) const
 { 
-    return VMAP::VMapFactory::createOrGetVMapManager()->isInLineOfSight(GetId(), x1, y1, z1, x2, y2, z2)
-        && _dynamicTree.isInLineOfSight(x1, y1, z1, x2, y2, z2, phasemask);
+    if ((checks & LINEOFSIGHT_CHECK_VMAP)
+      && !VMAP::VMapFactory::createOrGetVMapManager()->isInLineOfSight(GetId(), x1, y1, z1, x2, y2, z2))
```
Comment (BarbzYHOOL) [inline src/server/game/Maps/Map.cpp]: is it on purpose? the linebreak

Comment (Viste) [reply src/server/game/Maps/Map.cpp]: for easy read

Comment (BarbzYHOOL) [reply src/server/game/Maps/Map.cpp]: as you wish, you can keep it, I just asked xD

**Comment by wetbrownsauce** [comment]
How to test?

**Comment by Viste** [comment]
and need test basic los checks for sure they are dont break

**Comment by FrancescoBorzi** [comment]
Merged in Staging since contains generic changes that are hard to test properly and might affect a wide range of things.

Staging will be merged into master after some testing with test players.

**Pull request merged**